Auto generated mesh groups are now marked read only

AMZN-stankowi opened the pull request:
## What does this PR do?

Mesh groups that are automatically generated and not manually added are now marked read only.

![image](https://user-images.githubusercontent.com/4838196/210402893-516b55b8-8d7c-4b1b-a248-2dbc0fc9408c.png)


## How was this PR tested?

Manual testing: Loading several different scene files in the scene settings UI, verified read only worked as expected.
Ran the existing automated FBX tests + scene settings UI tests, they all passed.
Ran the new scene settings UI test, it passed.

lmbr-pip commented:
@AMZN-stankowi Is this for https://github.com/o3de/o3de/issues/13646 ?

AMZN-stankowi commented:
@lmbr-pip loosely, this will make it more obvious where those mesh groups come from, but it won't make the drag and drop itself less confusing

amzn-rhhong commented on Code/Tools/SceneAPI/SceneCore/DataTypes/Rules/IReadOnlyRule.h:
```diff
@@ -0,0 +1,30 @@
+        namespace DataTypes
+        {
+            // Disables UI interaction for the node and children of the node with this rule.
+            class IReadOnlyRule : public IRule
```
I'm not really a fan of creating a inheriting class between IRules and derived class (e.g ISkinRules) just to modify the tool tips. 
What about just add the modifyToolTips function to IRules and let the derived class handle the actual part of tool tip modification?

AMZN-stankowi replied:
I originally didn't have an IReadOnlyRule class, but it caused some serialization and linker issues due to how the systems are setup. SceneAPI / SceneCore are divided into a few pieces that make it tough to get everything linking in the right places.

amzn-rhhong replied:
Ok I think I understand the code a little better now. It's not a "ReadOnly" Rule - instead it's basically a rule that makes whatever container holds it to become read only, correct?
Maybe there could be a better name for this? UnmodifiableRule?

AMZN-stankowi replied:
Sure, I've renamed it to that. I didn't rename it everywhere because it's using the ReadOnly system in the reflected property editor, but I renamed the rule.

amzn-rhhong commented:
With this change I think it's even more crucial to add instructions on how to turn off the proc prefab in the tool tips.
Also maybe more UX inputs on this. Personally I think we could do something like this:
![ui mockup procprefab](https://user-images.githubusercontent.com/69218254/208507357-0191015d-71e6-47d9-a82f-b68037891a66.jpg)

junbo75 commented on Code/Tools/SceneAPI/SceneCore/DataTypes/Rules/IReadOnlyRule.h:
```diff
@@ -0,0 +1,30 @@
+#pragma once
```
Nit: May worth moving this line after the copyright comment.

AMZN-stankowi replied:
Done, and I updated a bunch of other ones to prevent this copy & paste error from happening again.

AMZN-stankowi commented:
Converting back to a draft while I work through an automated test. I'm going to be updating the branch with some WIP changes before going on holiday break.

AMZN-stankowi commented:
> With this change I think it's even more crucial to add instructions on how to turn off the proc prefab in the tool tips. Also maybe more UX inputs on this. Personally I think we could do something like this: ![ui mockup procprefab](https://user-images.githubusercontent.com/69218254/208507357-0191015d-71e6-47d9-a82f-b68037891a66.jpg)

I've updated the description to mention how to remove the group, and I've removed the delete button from the disabled mesh groups.
